mlir/lib/AsmParser/AsmParserImpl.h

```diff
@@ -340,12 +340,16 @@ class AsmParserImpl : public BaseT {
   /// Parse a list of comma-separated items with an optional delimiter.  If a
   /// delimiter is provided, then an empty list is allowed.  If not, then at
   /// least one element will be parsed.
-  ParseResult parseCommaSeparatedList(Delimiter delimiter,
-                                      function_ref<ParseResult()> parseElt,
-                                      StringRef contextMessage) override {
-    return parser.parseCommaSeparatedList(delimiter, parseElt, contextMessage);
+  ParseResult parseCommaSeparatedList(
+      Delimiter delimiter, function_ref<ParseResult()> parseElt,
+      std::optional<function_ref<ParseResult()>> parseSuffix,
```

Comment:
Can you move `parseSuffix` to the end and make it `function_ref<ParseResult()> parseSuffix = nullptr`? `function_ref` is rarely used with `std::optional`. Also, maybe this function would not be needed at all then?

Reply:
The function is needed to parse the suffix, which is optional. I'm using a `function_ref` similar to what we do for `parseElt`.